registry/api/protocol/eureka/instance_handler.go

```diff
@@ -106,60 +106,28 @@ func (routes *Routes) registerInstance(w rest.ResponseWriter, r *rest.Request) {
 		return
 	}
 
-	// Get the instance ID
-	// In the old eureka client versions (1.1.x) the instance ID is NOT explicitly set in the request data,
-	// but it is part of the DatacenterInfo class.
-	iid := inst.ID
-	if iid == "" {
-		iid, err = getInstanceID(inst)
-		if err != nil {
-			routes.logger.WithFields(log.Fields{
-				"namespace": r.Env[env.Namespace],
-				"error":     err,
-			}).Warnf("Failed to register instance %+v", inst)
-
-			i18n.Error(r, w, http.StatusBadRequest, i18n.ErrorInstanceRegistrationFailed)
-			return
-		}
-	}
-	uid := buildUniqueInstanceID(inst.Application, iid)
-
-	ttl := defaultDurationInt
-	if inst.Lease != nil && inst.Lease.DurationInt > 0 {
-		ttl = inst.Lease.DurationInt
-	}
-
-	catalog := routes.catalog(w, r)
-	if catalog == nil {
+	si, err := BuildA8Instance(inst)
+	if err != nil {
 		routes.logger.WithFields(log.Fields{
 			"namespace": r.Env[env.Namespace],
-			"error":     "catalog is nil",
-		}).Errorf("Failed to register instance %+v", inst)
+			"error":     err,
+		}).Warnf("Failed to register instance %+v", inst)
```

Comment:
why changing to warning instead of error? Because it's a user attributed to the user input?